process-chunk.js

```diff
@@ -1,16 +1,23 @@
 import { createHmac } from 'node:crypto'
 
-const generateSignature = function (content, secret) {
-  return createHmac('sha256', secret)
+const DIGEST_ALGORITHM = {
+  HS256: 'sha256',
+  HS384: 'sha384',
+  HS512: 'sha512'
+}
+
+const signatureGenerator = (algorithm, content) => (secret) =>
+  createHmac(DIGEST_ALGORITHM[algorithm], secret)
     .update(content)
     .digest('base64')
     .replace(/=/g, '')
     .replace(/\+/g, '-')
     .replace(/\//g, '_')
-}
-process.on('message', function ({ chunk, content, signature }) {
+
+process.on('message', function ({ chunk, content, signature, algorithm }) {
+  const generateSignature = signatureGenerator(algorithm, content)
```

Comment:
This function generator (currying) improve speed as we won’t pass each time `algorithm` and `content` during the loop (only the changing part)